hack/jenkins/README.md

```diff
@@ -0,0 +1,28 @@
+# Jenkins
+
+[Jenkins](http://jenkins-ci.org/) is a pluggable continuous
+integration system. The Google team is running a Jenkins server on a
+private GCE instance for the Kubernetes project in order to run longer
+integration tests, continuously, on different providers. Currently, we
+(Google) are only running Jenkins on our own providers (GCE and GKE)
+in different flavors.
+
+## General flow
+The flow of the Google Jenkins server:
+* Under the `kubernetes-build` job: Every 5 minutes, Jenkins polls for a batch of new commits, after which it runs the `build.sh` script (in this directory) on the latest tip. This results in build assets getting pushed to GCS and the `latest.txt` file in the `ci` bucket being updated. That job then triggers `kubernetes-e2e-*`.
+* On trigger, and every half hour (which effectively means all the time, unless we're failing cluster creation), e2e variants run, on the latest build assets in GCS:
+  * `kubernetes-e2e-gce`: Standard GCE e2e
+  * `kubernetes-e2e-gke`: GKE provider e2e, with head k8s and GKE creating clusters at its default version
+  * `kubernetes-e2e-gke-ci`: GKE provider e2e, with head k8s and GKE creating clusters at the head k8s version
+* Each named job may run in parallel with the others, i.e. `kubernetes-build` with `kubernetes-e2e-gce`, but `kubernetes-e2e-gce` will not run concurrently with itself.
```

Comment:
you use "but" here but this seems like orthogonal issues
- can a job run in parallel with iteself?
- can a job run in parallel with other jobs?

Maybe rewrite as "Each named job may run in parallel with itself and with the other jobs, except for 'kubernetes-e2e-gce' which cannot run in parallel with itself"

So IIUC, you may have an arbitrary number of kubernetes-build and kubrentes-e2e-gce tests running at the same time (for example, 5 if I do 5 commits within a half-second of each other) but only one kubernetes-e2e-gce at a time so in that example you will queue up 4 while the first one runs, and then drain the queue one at a time?